File: Data Structures & Algorithms/remove-element/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} val
     * @return {number}
     */
    removeElement(nums, val) {
        let w = 0
        for(let r = 0; r < nums.length; r++) {
            if(nums[r] != val) {
                nums[w] = nums[r];
                w++;
            }
        }
        return w;
    }
}

File: Data Structures & Algorithms/combination-target-sum/submission-0.py
class Solution:
    def combinationSum(self, nums: List[int], target: int) -> List[List[int]]:
        N = len(nums)
        ans = []

        def dfs(i: int, cur: List[int], s: int) -> None:
            if i >= N or s > target:
                return
            
            if s == target:
                ans.append(cur.copy())
                return
            
            cur.append(nums[i])
            dfs(i, cur, s + nums[i])
            cur.pop()
            dfs(i + 1, cur, s)

        dfs(0, [], 0)
        return ans